src/Npgsql/NpgsqlConnector.cs

```diff
@@ -648,25 +646,26 @@ async Task RawOpen(NpgsqlTimeout timeout, bool async, CancellationToken cancella
             }
             catch
             {
-#pragma warning disable CS8625
-                try { _stream?.Dispose(); } catch {
-                    // ignored
-                }
-                _stream = null;
-                try { _baseStream?.Dispose(); }
-                catch
-                {
-                    // ignored
-                }
-                _baseStream = null;
-                try { _socket?.Dispose(); }
-                catch
+                DisposeSafe(ref _stream);
+                DisposeSafe(ref _baseStream);
+                DisposeSafe(ref _socket);
+
+                void DisposeSafe<T>(ref T disposable)
+                    where T : class, IDisposable
                 {
-                    // ignored
+                    try
+                    {
+                        disposable?.Dispose();
+                        disposable = default!;
+                    }
+                    catch
+                    {
+                        // ignored
```

Comment:
Is this necessary? Dispose is never supposed to throw exceptions in .NET (and definitely not from these BCL types), and our current pattern (everywhere in the codebase) is to assume that (see NpgsqlConnector.Cleanup).

Reply:
Ah, just noticed that it was already this way. I'm fine with your new code, even if I'm not sure why we're doing this :)

Reply:
That's a good question I forgot to ask you. Why are there try-catches?

Reply:
I honestly don't know... Maybe I wrote these at a point where I didn't know Dispose shouldn't throw? Let me take a quick look at NetworkStream.Dispose

Reply:
I took a quick look. I don't see a clear thing showing that Dispose on NetworkStream/Socket will never throw - even if I think it should be considered a bug if they do. I guess we can leave the exception swallowing because why not... What do you think?

Reply:
Okay, will cleanup that code.

Reply:
Up to you - I'm frankly OK either way.